### ISSUE TITLE
incompatibility with foreach/YSI

### ISSUE THREAD

**Issue by marcelog09** [CyberMor/sampvoice#46]
I noticed that it doesn't work in my gamemode - only in filterscripts - and after some tests, I found out that there is some kind of incompatibility with foreach and / or YSI library.

In gamemode, when I put #include <sampvoice> before #include <foreach>, I get the following errors:

include \ sampvoice.inc (72): error 025: function heading differs from prototype
sampvoice.inc (73): error 025: function heading differs from prototype
gamemodes \ voicetest.pwn (14): error 025: function heading differs from prototype
gamemodes \ voicetest.pwn (24): error 025: function heading differs from prototype

Error lines:
forward SV_VOID: OnPlayerActivationKeyPress (SV_UINT: playerid, SV_UINT: keyid);
forward SV_VOID: OnPlayerActivationKeyRelease (SV_UINT: playerid, SV_UINT: keyid);

public SV_VOID: OnPlayerActivationKeyPress (
public SV_VOID: OnPlayerActivationKeyRelease (

And when placing #include <sampvoice> after #include <foreach>, no error is displayed, but the voice does not work - voices do not come out.

In filterscript - where there is no use of the foreach and / or YSI library - everything works correctly.

sampvoice version: 3.0-alpha
YSI version: YSI-5.4.102

Is there any solution?

**Comment by nrgrpg** [comment]
Same Here! 
Anyone can help us?

**Comment by Silvaa617** [comment]
Same problem...

**Comment by LegendaryNickBR** [comment]
here too... same problem. YSI v5.05.0505

**Comment by raedman90** [comment]
did anyone manage to solve it?

**Comment by SugengGod** [comment]
Is there any solution?

**Comment by Kutuy** [comment]
Any solution? We got same problem

**Comment by iGuihP** [comment]
> Qualquer solução? Temos o mesmo problema

Check if there is an include "Pawn.RakNet" in the .pwn, if there is: include the sampvoice AFTER the Pawn.RakNet :)

**Comment by mhyunata** [comment]
yeah we are same here, also i've test it with planting the script on my GM. but there's a many bugs, just like when u hit the talk button the sampvoice script will call another script from your GM, idk why but yeah it seems like this is cannot use with YSI/foreach
